### DISCUSSION TITLE
面板的数据库文件未加密是否具有泄露的风险

### DISCUSSION THREAD

**Discussion by cktk** [1Panel-dev/1Panel#5330] [Ideas]
### 1Panel 版本

1.10.7-lts

### 请描述您的需求或者改进建议

1Panel.db 这个文件  可以直接打开 没有加密  
如果系统遭遇后门或者自己程序有bug遭到这个文件的泄露  是否会对服务器产生更大的危害
比如我自己的程序有个任意目录读取的bug  

### 请描述你建议的实现方案

对db文件进行加密   加密的秘钥可以通过一定的算法生成 可以参考一下微信
密钥别丢日志文件里就行

**Comment by wanghe-fit2cloud** [comment]
v2.0 版本可能会考虑上述需求。
数据库文件加密会要求开启 CGO，这样就会存在不同操作系统的兼容问题，导致用户在安装阶段就会失败。
